src/neo/SmartContract/Native/Votes/VoteContract.cs

```diff
@@ -0,0 +1,428 @@
+            if (candidate.GetCandidate() == 0 || candidate.GetCandidate() > voteLength) return false;
+            VoteState voteState = new VoteState(voter, candidate);
+            return (AddVote(engine.Snapshot, voteState, id));
+        }
+
+        [ContractMethod(0_01000000, ContractParameterType.ByteArray)]
+        private StackItem GetVoteDetails(ApplicationEngine engine, VMArray args)
+        {
+            UInt256 TxHash = new UInt256(args[0].GetSpan());
+            var id = Crypto.Hash160(VoteCreateState.ConcatByte(TxHash.ToArray(), (engine.CallingScriptHash ?? this.Hash).ToArray()));
+            StorageKey create_key = CreateStorageKey(Prefix_CreateVote, id.ToArray());
+            StorageItem create_state = engine.Snapshot.Storages.TryGet(create_key);
+            return create_state.Value;
+        }
+
+        [ContractMethod(0_01000000, ContractParameterType.ByteArray, SafeMethod = true)]
+        private StackItem GetMultiStatistic(ApplicationEngine engine, VMArray args)
+        {
+            if (args[0] is null) return false;
+            UInt256 TxHash = new UInt256(args[0].GetSpan());
+            //Get Vote id
+            var id = Crypto.Hash160(VoteCreateState.ConcatByte(TxHash.ToArray(), (engine.CallingScriptHash ?? this.Hash).ToArray()));
+            StorageKey create_key = CreateStorageKey(Prefix_CreateVote, id.ToArray());
+            StorageItem create_byte = engine.Snapshot.Storages.TryGet(create_key);
+            VoteCreateState createState = Neo.IO.Helper.AsSerializable<VoteCreateState>(create_byte.Value);
+            if (!createState.IsSequence || !InteropService.Runtime.CheckWitnessInternal(engine, createState.Originator)) return false;
+            StorageKey index_key = CreateStorageKey(Prefix_Vote, id.ToArray());
+            byte[] prefix_key = StorageKey.CreateSearchPrefix(Hash, new[] { Prefix_Vote });
+            IEnumerable<(StorageKey, StorageItem)> pairs = engine.Snapshot.Storages.Find(prefix_key).OrderByDescending(p => p.Key);
+            if (pairs.Count() == 0) return false;
+            MultiStatistic result = new MultiStatistic();
+            List<UInt160> voteList = new List<UInt160>();
+            foreach (var pair in pairs)
+            {
+                VoteState vote_state = Neo.IO.Helper.AsSerializable<VoteState>(pair.Item2.Value);
+                UInt160 voter = vote_state.GetVoter();
+                if (voteList.Contains(voter)) continue;
+                voteList.Add(voter);
+                int account_balance = (int)new NeoToken().BalanceOf(engine.Snapshot, voter);
+                MultiCandidate candidate = Neo.IO.Helper.AsSerializable<MultiCandidate>(vote_state.ToArray());
+                result.AddVote(new CalculatedMultiVote(account_balance, candidate.GetCandidate()));
+            }
+            result.CalculateResult(new SchulzeModel());
+            var resultMatrix = result.ShowResult();
```

Comment:
As `resultMatrix` is a `byte` array type, it need to be parsed for a contract again. 
Better to convert to `VMArray[int][int]` type?